### DISCUSSION TITLE
feat: Display download progress in browser tab when downloading a model & switched to another tab

### DISCUSSION THREAD

**Discussion by silentoplayz** [open-webui/open-webui#1541] [General]
**Is your feature request related to a problem? Please describe.**
As a user of Open WebUI, I often find myself switching tabs while waiting for a model to download/be pulled from Open WebUI, directly from Ollama.com or via a HuggingFace GGUF model download URL. However, I have no way of knowing how much longer the download will take or whether it's still progressing while not actively on the Open WebUI page. This can be frustrating, especially if the download takes a long time to complete.

**Describe the solution you'd like**
I would like to see a visual indication of the download progress in the tab when I switch away from Open WebUI while a model is downloading/being pulled. This could be a progress bar or some other type of indicator that shows me the status of the download. This would allow me to easily see how much of the model has been downloaded and how long it might take to finish, even if I'm not actively viewing the Open WebUI tab.

**Describe alternatives you've considered**
One alternative I've considered is periodically switching back to the Open WebUI tab to check the download status. However, this can prove to be disruptive, as it interrupts the user's workflow. Another alternative is to receive a desktop notification when the download is complete. This could be a useful secondary indicator of the download status, but it doesn't provide the same level of visibility as a progress indication in the tab itself. And while a download completed notification already exists to solve the problem, it does not provide the user with the granularity of information that progress information for downloading a model in the browser tab would offer.

**Additional context**
This feature request relates to improving the user experience when downloading models from Open WebUI. Currently, there is no visual indication of the download progress, which can be frustrating for users. By adding a progress bar or other visual indication to the tab, users can easily see the status of the download and better plan their time accordingly.

Mockup Example: 
![DownloadProgress](https://github.com/open-webui/open-webui/assets/50341825/6577d272-ef44-4739-b701-ec9556b9457a)

**Comment by tjbck** [comment]
Interesting suggestion, but we do have a notification feature which would alert users when the model is downloaded. I'd love to hear what others think of this feature request as well.
